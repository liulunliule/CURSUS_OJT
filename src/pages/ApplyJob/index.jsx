import React from "react";
import "./index.scss";
import { Link } from "react-router-dom";

const ApplyJob = () => {
  return (
    <div className="apply-job">
      <div className="apply-job-title">
        <h1>Data Engineer</h1>
        <Link to="/secondLayout/careers" className="back-to-career">
          Back to Career
        </Link>
      </div>

      <div className="job-content">
        <button type="apply" className="apply-btn">Apply For This Job</button>
        <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur volutpat maximus pellentesque. Integer sem enim, luctus at nibh at, condimentum sagittis sapien. Sed tempus ipsum erat, sit amet efficitur velit interdum eu. Vestibulum hendrerit id dolor eu scelerisque. Phasellus ex dui, consequat nec feugiat eu, dapibus eget ante. Sed sodales interdum dui, at euismod mi feugiat hendrerit. Suspendisse auctor libero in tempor mollis. Nulla et dolor velit. Aliquam sit amet luctus quam.
        </p>
        <h2>What you'll be doing</h2>
        <p>
        Nam a egestas libero, eget eleifend turpis. Sed id ipsum a ipsum aliquam laoreet sit amet sit amet nibh. Proin dapibus, libero sed posuere rhoncus, orci mi cursus enim, at accumsan eros massa lacinia mi. Nunc eget finibus felis, volutpat malesuada sem. Aliquam ac nisl pellentesque, varius neque sit amet, porttitor nunc. Nullam elit tellus, dapibus non eleifend sed, hendrerit eget velit. Aliquam ut felis dictum, tincidunt magna vitae, aliquam massa. In porttitor tristique quam, non dignissim sapien pharetra ultrices. Cras non ante non velit mollis mollis. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Quisque et bibendum urna, eget consequat sapien. Integer sed condimentum nibh. Integer id neque tristique, lobortis massa ac, dapibus nibh. Donec nulla odio, porttitor ac rutrum eget, volutpat a velit. Curabitur et enim quis diam congue dictum et vitae dui. Nulla tortor orci, luctus a pretium vel, ultrices porta nisl.
        </p>
        <p>
        Etiam lobortis dictum tincidunt. Interdum et malesuada fames ac ante ipsum primis in faucibus. Suspendisse ultricies efficitur dui, suscipit tempus elit condimentum quis. Duis sed vestibulum tortor, eget cursus odio.
        </p>
        <h2>What we're looking for</h2>
          <p>1. Work closely with other data engineers to enhance infrastructure, improve reliability, and strengthen database performance</p>
          <p>2. Collaborate with data scientists, product managers, marketers and other data-driven internal stakeholders to identify and develop opportunities that enhance data access and usage</p>
          <p>3. Contribute to the ongoing development of the data warehouse ecosystem</p>
          <p>4. Become the expert on all aspects of Edututs+'s data reporting and analytics infrastructure</p>
          <p>5. Develop and maintain data definitions for internally available data sources</p>
        <h2>Who should apply?</h2>
        <p>
        Quisque et bibendum urna, eget consequat sapien. Integer sed condimentum nibh. Integer id neque tristique, lobortis massa ac, dapibus nibh. Donec nulla odio, porttitor ac rutrum eget, volutpat a velit. Curabitur et enim quis diam congue dictum et vitae dui. Nulla tortor orci, luctus a pretium vel, ultrices porta nisl.
        </p>
        <h2>What next?</h2>
        <p>
        Quisque et bibendum urna, eget consequat sapien. Integer sed condimentum nibh. Integer id neque tristique, lobortis massa ac, dapibus nibh. Donec nulla odio, porttitor ac rutrum eget, volutpat a velit. Curabitur et enim quis diam congue dictum et vitae dui. Nulla tortor orci, luctus a pretium vel, ultrices porta nisl.
        </p>
      </div>

      <form className="application-form">
        <h3>Submit Your Application</h3>
        <div className="form-group">
          <label>Full Name</label>
          <input type="text" />
        </div>
        <div className="form-group">
          <label>Email Address</label>
          <input type="email" />
        </div>
        <div className="form-group">
          <label>Phone Number</label>
          <input type="tel" />
        </div>
        <div className="form-group">
          <label>Gender</label>
          <select>
            <option value="">Select Gender</option>
            <option value="male">Male</option>
            <option value="female">Female</option>
          </select>
        </div>
        <div className="form-group">
          <label>Resume/CV</label>
          <input type="file" />
        </div>
        <div className="form-group">
          <label>Links</label>
          <input type="text" placeholder="LinkedIn URL" />
          <input type="text" placeholder="X URL" />
          <input type="text" placeholder="Portfolio URL" />
          <input type="text" placeholder="Other Website URL" />
        </div>
        <div className="form-group">
          <label>Additional Information</label>
          <textarea placeholder="Add a cover letter or anything else you want to share."></textarea>
        </div>
        <button type="submit" className="submit-btn">Submit Application</button>
      </form>
    </div>
  );
};

export default ApplyJob;