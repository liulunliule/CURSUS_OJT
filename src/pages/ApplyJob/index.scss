.apply-job {
    padding: 20px;
    max-width: 830px;
    margin: 0 auto;
    p {
      color: gray;
      margin-bottom: 30px;
    }
  
    .apply-job-title {
      display: flex;
      justify-content: space-between;
      align-items: center;
      margin-bottom: 20px;
      margin-top: 85px;
  
      h1 {
        font-size: 2em;
        margin: 0;
      }
  
      .back-to-career {
        background-color: red;
        color: #fff;
        padding: 10px 20px;
        text-decoration: none;
        border-radius: 5px;
        transition: background-color 0.3s;
  
        &:hover {
          background-color: black;
        }
      }
    }
  
    .job-content {
      .apply-btn {
        background-color: red;
        color: #fff;
        padding: 10px 20px;
        text-decoration: none;
        border-radius: 5px;
        border: none;
        cursor: pointer;
        font-size: 1em;
        transition: background-color 0.3s;
        margin-bottom: 20px;
  
        &:hover {
          background-color: black;
        }
      }

      p, ul {
        font-size: 1em;
        line-height: 1.6;
        margin-bottom: 20px;
      }
  
      h2 {
        font-size: 1.4em;
        margin: 20px 0 10px;
      }
  
      ul {
        list-style: disc;
        margin-left: 20px;
      }
    }
  
    .application-form {
      h3 {
        font-size: 1.6em;
        margin-bottom: 20px;
      }
  
      .form-group {
        margin-bottom: 20px;
  
        label {
          display: block;
          margin-bottom: 10px;
          font-weight: bold;
        }
  
        input, select, textarea {
          width: 100%;
          padding: 10px;
          font-size: 1em;
          border: 1px solid #ccc;
          border-radius: 5px;
          margin-bottom: 20px;
        }
  
        input[type="file"] {
          padding: 35px;
          border-radius: 4px;
        }
  
        textarea {
          resize: vertical;
          height: 100px;
        }
      }
  
      .submit-btn {
        background-color: red;
        color: #fff;
        padding: 10px 20px;
        text-decoration: none;
        border-radius: 5px;
        border: none;
        cursor: pointer;
        font-size: 1em;
        transition: background-color 0.3s;
  
        &:hover {
          background-color: black;
        }
      }
    }
  }  