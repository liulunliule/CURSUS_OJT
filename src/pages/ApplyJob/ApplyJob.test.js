import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { MemoryRouter } from 'react-router-dom';
import ApplyJob from '.';

const renderComponent = () =>
  render(
    <MemoryRouter>
      <ApplyJob />
    </MemoryRouter>
  );

describe('ApplyJob component', () => {
  test('renders ApplyJob component correctly', () => {
    const { asFragment } = renderComponent();
    expect(asFragment()).toMatchSnapshot();
  });

  test('displays the job title and back link', () => {
    renderComponent();
    const jobTitle = screen.getByText('Data Engineer');
    const backLink = screen.getByText('Back to Career');

    expect(jobTitle).toBeInTheDocument();
    expect(backLink).toBeInTheDocument();
    expect(backLink).toHaveAttribute('href', '/secondLayout/careers');
  });

  test('displays the apply button and job content', () => {
    renderComponent();
    const applyButton = screen.getByText('Apply For This Job');
    const jobContent = screen.getByText('Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur volutpat maximus pellentesque. Integer sem enim, luctus at nibh at, condimentum sagittis sapien. Sed tempus ipsum erat, sit amet efficitur velit interdum eu. Vestibulum hendrerit id dolor eu scelerisque. Phasellus ex dui, consequat nec feugiat eu, dapibus eget ante. Sed sodales interdum dui, at euismod mi feugiat hendrerit. Suspendisse auctor libero in tempor mollis. Nulla et dolor velit. Aliquam sit amet luctus quam.');

    expect(applyButton).toBeInTheDocument();
    expect(jobContent).toBeInTheDocument();
  });

  test('displays the sections for job description, requirements, and application form', () => {
    renderComponent();

    const sections = [
      'What you\'ll be doing',
      'What we\'re looking for',
      'Who should apply?',
      'What next?',
      'Submit Your Application',
    ];

    sections.forEach(section => {
      expect(screen.getByText(section)).toBeInTheDocument();
    });
  });

  test('displays the application form with correct fields', () => {
    renderComponent();

    const formFields = [
      'Full Name',
      'Email Address',
      'Phone Number',
      'Gender',
      'Resume/CV',
      'Links',
      'Additional Information',
    ];

    formFields.forEach(field => {
      expect(screen.getByText(field)).toBeInTheDocument();
    });

    expect(screen.getByPlaceholderText('LinkedIn URL')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('X URL')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Portfolio URL')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Other Website URL')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Add a cover letter or anything else you want to share.')).toBeInTheDocument();
  });

  test('submit button is present in the application form', () => {
    renderComponent();
    const submitButton = screen.getByText('Submit Application');
    expect(submitButton).toBeInTheDocument();
  });

  test('form submission triggers correctly', () => {
    renderComponent();
    const submitButton = screen.getByText('Submit Application');

    fireEvent.click(submitButton);

    expect(submitButton).toBeInTheDocument();
  });
});