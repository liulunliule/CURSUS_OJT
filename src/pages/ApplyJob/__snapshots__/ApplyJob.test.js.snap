// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplyJob component renders ApplyJob component correctly 1`] = `
<DocumentFragment>
  <div
    class="apply-job"
  >
    <div
      class="apply-job-title"
    >
      <h1>
        Data Engineer
      </h1>
      <a
        class="back-to-career"
        href="/secondLayout/careers"
      >
        Back to Career
      </a>
    </div>
    <div
      class="job-content"
    >
      <button
        class="apply-btn"
        type="apply"
      >
        Apply For This Job
      </button>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur volutpat maximus pellentesque. Integer sem enim, luctus at nibh at, condimentum sagittis sapien. Sed tempus ipsum erat, sit amet efficitur velit interdum eu. Vestibulum hendrerit id dolor eu scelerisque. Phasellus ex dui, consequat nec feugiat eu, dapibus eget ante. Sed sodales interdum dui, at euismod mi feugiat hendrerit. Suspendisse auctor libero in tempor mollis. Nulla et dolor velit. Aliquam sit amet luctus quam.
      </p>
      <h2>
        What you'll be doing
      </h2>
      <p>
        Nam a egestas libero, eget eleifend turpis. Sed id ipsum a ipsum aliquam laoreet sit amet sit amet nibh. Proin dapibus, libero sed posuere rhoncus, orci mi cursus enim, at accumsan eros massa lacinia mi. Nunc eget finibus felis, volutpat malesuada sem. Aliquam ac nisl pellentesque, varius neque sit amet, porttitor nunc. Nullam elit tellus, dapibus non eleifend sed, hendrerit eget velit. Aliquam ut felis dictum, tincidunt magna vitae, aliquam massa. In porttitor tristique quam, non dignissim sapien pharetra ultrices. Cras non ante non velit mollis mollis. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Quisque et bibendum urna, eget consequat sapien. Integer sed condimentum nibh. Integer id neque tristique, lobortis massa ac, dapibus nibh. Donec nulla odio, porttitor ac rutrum eget, volutpat a velit. Curabitur et enim quis diam congue dictum et vitae dui. Nulla tortor orci, luctus a pretium vel, ultrices porta nisl.
      </p>
      <p>
        Etiam lobortis dictum tincidunt. Interdum et malesuada fames ac ante ipsum primis in faucibus. Suspendisse ultricies efficitur dui, suscipit tempus elit condimentum quis. Duis sed vestibulum tortor, eget cursus odio.
      </p>
      <h2>
        What we're looking for
      </h2>
      <p>
        1. Work closely with other data engineers to enhance infrastructure, improve reliability, and strengthen database performance
      </p>
      <p>
        2. Collaborate with data scientists, product managers, marketers and other data-driven internal stakeholders to identify and develop opportunities that enhance data access and usage
      </p>
      <p>
        3. Contribute to the ongoing development of the data warehouse ecosystem
      </p>
      <p>
        4. Become the expert on all aspects of Edututs+'s data reporting and analytics infrastructure
      </p>
      <p>
        5. Develop and maintain data definitions for internally available data sources
      </p>
      <h2>
        Who should apply?
      </h2>
      <p>
        Quisque et bibendum urna, eget consequat sapien. Integer sed condimentum nibh. Integer id neque tristique, lobortis massa ac, dapibus nibh. Donec nulla odio, porttitor ac rutrum eget, volutpat a velit. Curabitur et enim quis diam congue dictum et vitae dui. Nulla tortor orci, luctus a pretium vel, ultrices porta nisl.
      </p>
      <h2>
        What next?
      </h2>
      <p>
        Quisque et bibendum urna, eget consequat sapien. Integer sed condimentum nibh. Integer id neque tristique, lobortis massa ac, dapibus nibh. Donec nulla odio, porttitor ac rutrum eget, volutpat a velit. Curabitur et enim quis diam congue dictum et vitae dui. Nulla tortor orci, luctus a pretium vel, ultrices porta nisl.
      </p>
    </div>
    <form
      class="application-form"
    >
      <h3>
        Submit Your Application
      </h3>
      <div
        class="form-group"
      >
        <label>
          Full Name
        </label>
        <input
          type="text"
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Email Address
        </label>
        <input
          type="email"
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Phone Number
        </label>
        <input
          type="tel"
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Gender
        </label>
        <select>
          <option
            value=""
          >
            Select Gender
          </option>
          <option
            value="male"
          >
            Male
          </option>
          <option
            value="female"
          >
            Female
          </option>
        </select>
      </div>
      <div
        class="form-group"
      >
        <label>
          Resume/CV
        </label>
        <input
          type="file"
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Links
        </label>
        <input
          placeholder="LinkedIn URL"
          type="text"
        />
        <input
          placeholder="X URL"
          type="text"
        />
        <input
          placeholder="Portfolio URL"
          type="text"
        />
        <input
          placeholder="Other Website URL"
          type="text"
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Additional Information
        </label>
        <textarea
          placeholder="Add a cover letter or anything else you want to share."
        />
      </div>
      <button
        class="submit-btn"
        type="submit"
      >
        Submit Application
      </button>
    </form>
  </div>
</DocumentFragment>
`;